force-app/main/default/classes/Trigger Recipes/MetadataTriggerHandler.cls

```diff
@@ -0,0 +1,131 @@
+/**
+ * @description This class exists as a unified, trigger handler class. It
+ * uses Custom Metadata, and introspection of the Trigger.new variable to
+ * determine what trigger handler classes should be called, and in what order.
+ *
+ * Metadata_Driven_Trigger__mdt has three fields:
+ *  * Object__c - is a metadata entity look up to an sObject ie: Account
+ *  * Execution_Order__c - is an integer and determines the order the trigger
+ *  *   handlers are executed
+ *  * Class__c - is a String holding the name of the Trigger Handler to execute
+ *
+ * Note: This Trigger framework works like this:
+ *
+ * An .trigger for a sObject invokes this class via:
+ * new MetadataTriggerHandler().run();
+ *
+ * This trigger handler class extends TriggerHandler - all the trigger handler
+ * classes _must_ extend trigger handler. Most classes will only overwrite the
+ * context methods like afterUpdate(). This class, however, overrides the run
+ * method. This class is responsible for determining which other trigger
+ * handler classes to instantiate and run.
+ *
+ * Concrete example:
+ * AccountTrigger.trigger (in this org) - invokes this class.
+ * This class queries the custom metadata and will find (at least) one metadata
+ * record tied to Account and the metadata record's Class__c specifies
+ * AccountTriggerHandler. This class then loops over the returned metadata
+ * records, instantiating the classes specified. It then calls the appropriate
+ * context methods on those classes.
+ *
+ */
+public with sharing class MetadataTriggerHandler extends TriggerHandler {
+    /**
+     * @description Overrides the standard Run() method, which allows this
+     * metadata based trigger handler can be an incremental update / sit beside
+     * other trigger handlers classes that are directly invoked by a trigger
+     */
+    override public void run() {
+        /**
+         * This is standard validation checking for triggers / triggerhandlers
+         */
+        if (!validateRun()) {
+            return;
+        }
+
+        addToLoopCount();
+
+        /**
+         * End standard validation
+         */
+
+        String objType = getSobjectType();
+        /**
+         * This query finds an ordered list trigger handler classes
+         * to execute. It ignores any classes that are marked as disabled.
+         *
+         * Note: It will exclude any triggerHandler metadata records for which
+         * the user's email address is found in a related disabled_for__mdt
+         * record.
+         *
+         * Admin's and Developers can selectively disable trigger handlers
+         * for all or selected individuals *without* deploying.
+         *
+         */
+        List<Metadata_Driven_Trigger__mdt> tryggers = [
```

Comment:
Flag this for PMD Exception else build will fail!

Reply:
I'd like us (not us as a team, but us as a company establish a finer-grained, less false-positive approach to enforcing fls/crud checks.

Reply:
Agreed! Going to do something about it soon :)